src/components/molecules/EventDisplay/EventDisplay.tsx

```diff
@@ -17,13 +17,15 @@ import "./EventDisplay.scss";
 
 interface EventDisplayProps {
   event: firebase.firestore.DocumentData;
-  venue: WithId<AnyVenue>;
+  venue?: WithId<AnyVenue>;
 }
 
 export const EventDisplay: FC<EventDisplayProps> = ({ event, venue }) => {
   const { user, profile } = useUser();
 
   const enterEvent = useCallback(() => {
+    if (!venue) return;
+
     const room = venue?.rooms?.find((room) => room.title === event.room);
```

Comment:
[nit] since we're now explicitly checking `venue` above, we shouldn't need the possibly `undefined` checks here anymore?

Reply:
Agreed; this is the sort of thing a linter can auto fix at scale.